docs/content/sips/0xx-component-dependencies.md

```diff
@@ -0,0 +1,108 @@
+
+# Proposal
+Developers can use the following syntax to specify that they want to use a component as a dependency, either from a local disk or via a remote component registry:
+
+## Using dependencies from disk (local)
+```toml
+[component."infra-dashboard".dependencies]
+# Components can satisfy dependencies by import name (e.g., "aws:client/s3")
+# using a component from disk. It is implicitly assumed that the
+# `my_aws_client.wasm` component exports an instance of the
+# `aws:client/s3` interface.
+"aws:client/s3" = { path = "my_aws_client.wasm" }
+
+# Optionally, explicitly specify the export name to use to satisfy the `aws:client/s3` interface.
+"aws:client/s3" = { path = "my_aws_client.wasm", export = "my-s3-client" }
+
+# Without an interface name, attempt to resolve every import of the "aws:client" package.
+"aws:client" = { path = "my_aws_client.wasm" }
+```
+
+> ⚠️ NOTE: Using the package and interface forms together (e.g. `aws:client` and `aws:client/s3` respectively) within the same dependencies section will result in an error to prevent resolution ambiguities.
+
+## Using dependencies from a registry (remote)
+```toml
+[component."infra-dashboard".dependencies]
+# Equivalent to { version = "1.0.0" , package = "aws:client"}
+"aws:client" = "1.0.0"
+
+# Use the `aws:client` component package to satisfy any number of "wasi:blobstore" imports ...
+"wasi:blobstore" = { registry = "my-registry.io", version = "0.1.0", package = "aws:client" }
+```
+
+## Dependency Isolation
+Statically composing the parent Spin component and its dependencies yields a component where the parent component and dependencies are no longer isolated from eachother. Each dependency will inherit the configuration of the parent component, which may be desireable in some library use-cases (e.g., allowing the `aws:client/s3` dependency to make outbound requests via `allowed_outbound_hosts`). However, making this the default behavior contradicts the explicit opt-in nature of capabilities for Spin (and the component model), so by default, component dependencies in Spin will not inherit configuration.
+
+To allow dependencies to inherit configuration from the parent Spin component, developers can set `dependencies_inherit_configuration = true`, as shown in the following example:
+
+```toml
+[component."infra-dashboard"]
+# ...
+allowed_outbound_hosts = ["https://s3.us-west-2.amazonaws.com/my-bucket/puppy.jpeg"]
+dependencies_inherit_configuration = true
+
+[component."infra-dashboard".dependencies]
+"aws:client" = "1.0.0"
+```
+
+The above configuration allows the component used to satisfy the `aws:client` dependency to make outbound HTTP requests to the S3 bucket specified in `allowed_outbound_hosts`. If `dependencies_inherit_configuration` is omitted or set to `false`, an `AccessDenied` error will be returned whenever the `aws:client` component attempts to make an outbound HTTP request.
+
+### The `spin-deny-all` adapter
```

Comment:
It would be great with a small note (or reference to) what an 'adapter' is. I've heard the term used in the world of components before, and I think it may be just another component, but with a particular purpose?

Reply:
An adapter is just a kind of component that is designed to wrap around another component and change its imports/exports in some way.

Reply:
Side note: There is a slightly different (albeit related) usage of "adapter" which is the adapters used to wrap preview1 core modules in a preview 2 components (i.e., a component wrapping a module instead of the usage here of a component wrapping a component).